src/Shared/SqlProcessor.cs

```diff
@@ -35,6 +36,12 @@ internal static class SqlProcessor
     private static readonly SearchValues<char> WhitespaceSearchValues = SearchValues.Create(WhitespaceChars);
 #endif
 
+    // This is not an exhaustive list but covers the majority of common reserved SQL keywords that may follow a FROM clause.
+    // This is used when determining if the previous token is a keyword in order to identify the end of a comma separated FROM clause.
+    private static readonly string[] FromClauseReservedKeywords = [
+        "AS", "JOIN", "CROSS", "WHERE", "BY", "HAVING", "WINDOW", "LIMIT", "OFFSET", "TABLESAMPLE", "WITH", "PIVOT", "UNPIVOT"
```

Comment:
Nit: can we sort these? Makes it easier to see if something needs adding/is missing in the future.

Reply:
OK, in that case can you just add a comment to that effect?